Refactored Tater Effect registration and updated docs

Manchick0 opened the pull request:
Refactored Tater Effect to use RegistryEntry, since the StatusEffect itself isn't used much anymore. Updated related potion usage to reflect this change. Enhanced documentation with detailed instructions on applying effects programmatically.

Commit:
Refactored Tater Effect registration and updated docs

Refactored Tater Effect to use RegistryEntry, since the StatusEffect itself isn't used much anymore. Updated related potion usage to reflect this change. Enhanced documentation with detailed instructions on applying effects programmatically.

IMB11 commented:
Hi @Manchick0, would you be able to fix the markdown lint issues and spotless checkstyle issues for the reference mod? We'd be able to get this merged after it's done. Thanks :+1:

Commit:
Refactor effects documentation for clarity and structure

Updated effects section to improve readability and consistency. Introduced a table format for argument descriptions and refined terminology for better understanding.

Commit:
Satisfy the building process

Manchick0 commented:
> Hi @Manchick0, would you be able to fix the markdown lint issues and spotless checkstyle issues for the reference mod? We'd be able to get this merged after it's done. Thanks 👍

Of course! Already done!

Manchick0 committed:
Merge branch 'main' into main

IMB11 commented on develop/entities/effects.md:
````diff
@@ -57,10 +58,30 @@ language file.
+a `StatusEffectInstance`, and returns a boolean, specifying whether the effect was successfully applied.
+
+```java
+var instance = new StatusEffectInstance(FabricDocsReferenceEffects.TATER, 5 * 20, 0, false, true, true);
````
This should really be in the reference mod, even just in the entrypoint if nessecary

Manchick0 replied:
The thing is, where should I put it?

Manchick0 replied:
the only logical thing would be a static grantEffect() method, or something similar

IMB11 reviewed (changes requested):
Just 1 minor change and then it should be good to merge.

its-miroma commented on develop/entities/effects.md:
```diff
@@ -57,10 +58,30 @@ language file.
+
+::: tip
+For a quick test, it might be a better idea to use the previously mentioned `/effect` command: <br/>
+`/effect give @p fabric-docs-reference:tater`
```
```suggestion
For a quick test, it might be a better idea to use the previously mentioned `/effect` command:

```mcfunction
/effect give @p fabric-docs-reference:tater
\```
```

^^ Without the backslash though...

its-miroma replied:
The backslash slipped through

Manchick0 replied:
yeah I saw that

IMB11 committed:
Update develop/entities/effects.md

IMB11 committed:
Update develop/entities/effects.md

Co-authored-by: Miroma <136986257+its-miroma@users.noreply.github.com>

Commit:
Add ReferenceMethods class for version-aware method access

This commit introduces a static utility class to centralize version-aware internal method references, starting with `addTaterEffect`. Additionally, updated documentation now transcludes the new method for consistent and maintainable reference examples.

Commit:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	develop/entities/effects.md

Commit:
Satisfy the building process
